decavision/dataset_preparation/generate_pseudolabels.py

```diff
@@ -0,0 +1,159 @@
+import os
+import pandas as pd
+import tensorflow as tf
+import numpy as np
+import shutil
+import matplotlib.pyplot as plt
+
+from tensorflow.keras.models import load_model
+from distutils.dir_util import copy_tree
+from tqdm import tqdm
+
+
+class PseudoLabelGenerator:
+    """
+    Class to make pseudo labels.
+    """
+
+    def __init__(self):
+        pass
+
+    def _load_img(self, path, target_size=(299, 299)):
+        """
+        Load an image from a given path and normalize it
+
+        Args:
+            path (list): Input image path
+            target_size (tuple): Size of image
+        Returns:
+            np.array: Numpy array of the data
+        """
+        # Read image
+        bits = tf.io.read_file(path)
+        image = tf.image.decode_jpeg(bits, channels=3)
+        # Resize
+        image = tf.image.resize(image, size=[*target_size])
+        image = tf.reshape(image, [*target_size, 3])
+        image = tf.cast(image, tf.uint8)
+        # Normalize [0, 1]
+        image = tf.image.convert_image_dtype(image, dtype=tf.float32)
+        image = image.numpy()
+        return image
+
+    def generate_pseudolabel_data(self, model_path,
```

Comment:
Some of these arguments should be inputs to the class instead of this method I think. For example the paths because they could be used in other methods.

Reply:
Added all path arguments in `init`.